### PR TITLE
Open our own generated html files in the help pane of RStudio

### LINKED ISSUE #120: Supply macro definitions in print.dev_help
`print.dev_topic()` — the documentation suggests that `...` in `Rd2txt()` and `Rd2html()` is passed on to `parse_Rd()`.  I don't remember exactly what you need to supply in the `macros` argument, but I'm pretty sure I got it working in pkgdown.

Would also need to work with the IDE team to get the necessary information passed along to the `previewRd()` API call.

### PR DESCRIPTION
**Pull request by lionel-**
Closes #120

### CODE REVIEW

File: R/dev-help.R [hunk 1]
```diff
@@ -56,19 +56,32 @@ load_rd_macros <- function(dir) {
...
+    file.show(path, title = title)
   } else if (type == "html") {
+    topic_write_html(x, path)
+
     if (is_installed("rstudioapi") && rstudioapi::hasFun("previewRd")) {
-      rstudioapi::callFun("previewRd", x$path)
+      # This localhost URL is also part of getting RStudio to open in
+      # the help pane
+      port <- exec(env_get(ns_env("tools"), "httpdPort"))
+      url <- sprintf("http://localhost:%i/%s", port, doc_path)
```
Comment (kevinushey) [inline R/dev-help.R:79]: I think you could run this code even with `rstudioapi` isn't available -- you can check if this is an RStudio R session via something like `.Platform$GUI == "RStudio"`.

File: R/dev-help.R [hunk 2]
```diff
@@ -56,19 +56,32 @@ load_rd_macros <- function(dir) {
 print.dev_topic <- function(x, ...) {
   cli::cli_inform(c("i" = "Rendering development documentation for {.val {x$topic}}"))
 
-  type <- match.arg(x$type %||% "text", c("text", "html"))
-  out_path <- paste(tempfile("Rtxt"), type, sep = ".")
+  type <- arg_match0(x$type %||% "text", c("text", "html"))
+  file <- fs::path_ext_set(fs::path_file(x$path), type)
+
+  # This directory structure is necessary for RStudio to open the
+  # .html file in the help pane (see rstudio/rustdio#11336)
```
Comment (hadley) [inline R/dev-help.R:63]:
```suggestion
  # .html file in the help pane (see rstudio/rstudio#11336)
```

File: R/dev-help.R [hunk 3]
```diff
@@ -56,19 +56,32 @@ load_rd_macros <- function(dir) {
...
-    file.show(out_path, title = paste(x$pkg, basename(x$path), sep = ":"))
+    topic_write_text(x, path)
+    title <- paste(x$pkg, basename(x$path), sep = ":")
+    file.show(path, title = title)
   } else if (type == "html") {
+    topic_write_html(x, path)
+
     if (is_installed("rstudioapi") && rstudioapi::hasFun("previewRd")) {
-      rstudioapi::callFun("previewRd", x$path)
+      # This localhost URL is also part of getting RStudio to open in
```
Comment (hadley) [inline R/dev-help.R:76]: What's the advantage over `previewRd`?

**Review by hadley — APPROVED** [review]

File: R/dev-help.R (hunk 3 above)
Comment (kevinushey) [reply R/dev-help.R:76]: `previewRd` doesn't handle package-local macro definitions; see https://github.com/rstudio/rstudio/issues/11336

Comment (hadley) [reply R/dev-help.R:76]: Ah, got it.

File: R/dev-help.R (hunk 1 above)
Comment (lionel-) [reply R/dev-help.R:79]: Thanks, done.

**Pull request merged**